python/pylibcudf/pylibcudf/aggregation.pyi

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from enum import IntEnum, auto
+
+from pylibcudf.types import (
+    DataType,
+    Interpolation,
+    NanEquality,
+    NullEquality,
+    NullOrder,
+    NullPolicy,
+    Order,
+)
+
+class Kind(IntEnum):
+    SUM = auto()
+    PRODUCT = auto()
+    MIN = auto()
+    MAX = auto()
+    COUNT_VALID = auto()
+    COUNT_ALL = auto()
+    ANY = auto()
+    ALL = auto()
+    SUM_OF_SQUARES = auto()
+    MEAN = auto()
+    VARIANCE = auto()
+    STD = auto()
+    MEDIAN = auto()
+    QUANTILE = auto()
+    ARGMAX = auto()
+    ARGMIN = auto()
+    NUNIQUE = auto()
+    NTH_ELEMENT = auto()
+    RANK = auto()
+    COLLECT_LIST = auto()
+    COLLECT_SET = auto()
+    PTX = auto()
+    CUDA = auto()
+    CORRELATION = auto()
+    COVARIANCE = auto()
+
+class CorrelationType(IntEnum):
+    PEARSON = auto()
+    KENDALL = auto()
+    SPEARMAN = auto()
+
+class EWMHistory(IntEnum):
+    INFINITE = auto()
+    FINITE = auto()
+
+class RankMethod(IntEnum):
+    FIRST = auto()
+    AVERAGE = auto()
+    MIN = auto()
+    MAX = auto()
+    DENSE = auto()
+
+class RankPercentage(IntEnum):
+    NONE = auto()
+    ZERO_NORMALIZED = auto()
+    ONE_NORMALIZED = auto()
+
+class UdfType(IntEnum):
+    CUDA = auto()
+    PTX = auto()
+
+class Aggregation:
+    def kind(self) -> Kind: ...
+
+def sum() -> Aggregation: ...
+def product() -> Aggregation: ...
+def min() -> Aggregation: ...
+def max() -> Aggregation: ...
+def count(null_handling: NullPolicy = NullPolicy.INCLUDE) -> Aggregation: ...
```

Comment:
We can elide default values by doing `def count(null_handling: NullPolicy = ...)`.

However, type stubs are not just for type checking, but also LSP. In the latter case, having the default value encoded _is_ useful since I can immediately see from the signature what behaviour I get if I don't provide the argument.

I will see if I can automatically put the right value in here.

Reply:
For this round, I am not going to put defaults in automatically.